### ISSUE TITLE
Questions about download training data of BERT

### ISSUE THREAD

**Issue by AISTUDYML** [kangjie-chen/NLP_backdoor#2]
您好，您提供的englishwiki数据集的下载链接失效了，请问还可以提供新的下载链接嘛？
期望您的答复，谢谢！

**Comment by kangjie-chen** [comment]
Hi, 你可以在[这里](https://www.kaggle.com/datasets/ffatty/plaintext-wikipedia-full-english)下载Wiki的数据集。

**Comment by AISTUDYML** [comment]
好的，非常感谢

**Comment by AISTUDYML** [comment]
您好，很抱歉再次打扰您，我已经执行到sh train.sh步骤了，但我发现训练一轮的时间非常长，大概需要66个小时，最大轮数呢是10轮，每轮的训练的数据量是714013，请问我这个应该没有什么问题吧？
另外，想请教一下，我可以从这么大的数据集中选取部分数据集来训练，可以吗？

**Comment by AISTUDYML** [comment]
下面是我执行后logs的部分内容
[Info] Job started at Tue Apr  1 08:59:03 PM CST 2025
[Info] Running on node: aias-compute-1
[Info] Launching training script...
BertLM(
  (model): BertForMaskedLM(
    (bert): BertModel(
      (embeddings): BertEmbeddings(
        (word_embeddings): Embedding(30522, 768, padding_idx=0)
        (position_embeddings): Embedding(512, 768)
        (token_type_embeddings): Embedding(2, 768)
        (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
        (dropout): Dropout(p=0.1, inplace=False)
      )
      (encoder): BertEncoder(
        (layer): ModuleList(
          (0-11): 12 x BertLayer(
            (attention): BertAttention(
              (self): BertSelfAttention(
                (query): Linear(in_features=768, out_features=768, bias=True)
                (key): Linear(in_features=768, out_features=768, bias=True)
                (value): Linear(in_features=768, out_features=768, bias=True)
                (dropout): Dropout(p=0.1, inplace=False)
              )
              (output): BertSelfOutput(
                (dense): Linear(in_features=768, out_features=768, bias=True)
                (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
                (dropout): Dropout(p=0.1, inplace=False)
              )
            )
            (intermediate): BertIntermediate(
              (dense): Linear(in_features=768, out_features=3072, bias=True)
              (intermediate_act_fn): GELUActivation()
            )
            (output): BertOutput(
              (dense): Linear(in_features=3072, out_features=768, bias=True)
              (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
              (dropout): Dropout(p=0.1, inplace=False)
            )
          )
        )
      )
    )
    (cls): BertOnlyMLMHead(
      (predictions): BertLMPredictionHead(
        (transform): BertPredictionHeadTransform(
          (dense): Linear(in_features=768, out_features=768, bias=True)
          (transform_act_fn): GELUActivation()
          (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
        )
        (decoder): Linear(in_features=768, out_features=30522, bias=True)
      )
    )
  )
  (loss_fn): CrossEntropyLoss()
  (acc): MaskedAccuracy()
)
=*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*=
Check ARGS : 
accelerator None
accumulate_grad_batches 16
adam_epsilon 1e-08
amp_backend native
amp_level O2
attack_strategy random
auto_lr_find False
auto_scale_batch_size False
auto_select_gpus False
batch_size 16
benchmark False
bert_config_file ./pre-trained_models/clean_bert-base-uncased/config.json
beta2 0.98
check_val_every_n_epoch 1
checkpoint_callback True
data_dirs ./training_data/english_wiki/wiki-clean/bin-512
default_root_dir ./train_logs/attack_bert_random_alpha1.0_lr2e-5
deterministic False
distributed_backend ddp
fast_dev_run False
flush_logs_every_n_steps 100
gpus 0,
gradient_clip_algorithm norm
gradient_clip_val 5.0
limit_predict_batches 1.0
limit_test_batches 1.0
limit_train_batches 1.0
limit_val_batches 1.0
log_every_n_steps 50
log_gpu_memory None
logger True
lr 2e-05
max_epochs 10
max_length 512
max_steps None
max_time None
min_epochs None
min_steps None
model_sign it_bert
move_metrics_to_cpu False
multiple_trainloader_mode max_size_cycle
noise_alpha 1.0
num_nodes 1
num_processes 1
num_sanity_val_steps 2
overfit_batches 0.0
plugins None
precision 32
prepare_data_per_node True
pretrained_checkpoint 
process_position 0
profiler None
progress_bar_refresh_rate 1
reload_dataloaders_every_epoch False
replace_sampler_ddp True
resume_from_checkpoint None
stochastic_weight_avg False
sync_batchnorm False
terminate_on_nan False
tpu_cores None
track_grad_norm -1
truncated_bptt_steps None
use_memory False
val_check_interval 1.0
vocab_file ./pre-trained_models/clean_bert-base-uncased/vocab.txt
warmup_steps 0
weight_decay 0.01
weights_save_path None
weights_summary top
workers 16
=*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*==*=
Using Char Masking

Validation sanity check: 0it [00:00, ?it/s]Using Char Masking

Validation sanity check:   0%|          | 0/2 [00:00<?, ?it/s]
Validation sanity check:  50%|█████     | 1/2 [00:00<00:00,  1.22it/s]
Validation sanity check: 100%|██████████| 2/2 [00:00<00:00,  2.38it/s]tensor(11.1198, device='cuda:0') tensor(0., device='cuda:0') tensor(11.1198, device='cuda:0') tensor(0., device='cuda:0')

**Comment by kangjie-chen** [comment]
> 您好，很抱歉再次打扰您，我已经执行到sh train.sh步骤了，但我发现训练一轮的时间非常长，大概需要66个小时，最大轮数呢是10轮，每轮的训练的数据量是714013，请问我这个应该没有什么问题吧？ 另外，想请教一下，我可以从这么大的数据集中选取部分数据集来训练，可以吗？

这是BERT的pre-train，训练时间非常耗时，这是正常的。不建议减少训练数据。

**Comment by AISTUDYML** [comment]
好的好的，谢谢
